### PR TITLE
Fix app-compat differences in System.Net.Requests

### LINKED ISSUE #18151: Desktop: System.Net.Tests.FtpWebRequestTest.Ctor_VerifyDefaults_Success failed with "Xunit.Sdk.EqualException"
Failed test: System.Net.Tests.FtpWebRequestTest.Ctor_VerifyDefaults_Success

Detail: https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_netfx_windows_nt_debug/13/testReport/System.Net.Tests/FtpWebRequestTest/Ctor_VerifyDefaults_Success/

~~~
MESSAGE:
Assert.False() Failure\r\nExpected: False\r\nActual: True
+++++++++++++++++++
STACK TRACE:
at System.Net.Tests.FtpWebRequestTest.Ctor_VerifyDefaults_Success() in D:\j\workspace\outerloop_net---903ddde6\src\System.Net.Requests\tests\FtpWebRequestTest.cs:line 43
~~~

### PR DESCRIPTION
**Pull request by davidsh**
Fix some product code behavior differences between .NET Core
and .NET Framework (Desktop).

Adjust tests. Disable more tests not working on Desktop.

Fixes #18151.
Contributes to #17842.

### CODE REVIEW

**Comment by davidsh** [comment]
cc: @karelz

**Comment by davidsh** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu14.04 Debug

File: src/System.Net.Requests/src/System/Net/AuthenticationManager.cs [hunk 1]
```diff
@@ -27,6 +27,10 @@ public static Authorization PreAuthenticate(WebRequest request, ICredentials cre
 
         public static void Register(IAuthenticationModule authenticationModule)
         {
+            if (authenticationModule == null)
+            {
+                throw new ArgumentNullException();
```
Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/AuthenticationManager.cs]: Nit: `throw new ArgumentNullException(nameof(authenticationModule));`?

Comment (davidsh) [inline src/System.Net.Requests/src/System/Net/AuthenticationManager.cs]: Fixed.

**Comment by davidsh** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu14.04 Debug

File: src/System.Net.Requests/src/System/Net/FtpWebRequest.cs [hunk 2]
```diff
@@ -1576,7 +1576,7 @@ public bool KeepAlive
         {
             get
             {
-                return false;
```
Comment (CIPop) [inline src/System.Net.Requests/src/System/Net/FtpWebRequest.cs]: Does the protocol impl. actually have support for this?

File: src/System.Net.Requests/src/System/Net/HttpWebRequest.cs [hunk 3]
```diff
@@ -136,16 +136,20 @@ public string Accept
 
         public virtual bool AllowReadStreamBuffering
         {
+            // Same behavior as .NET Framework.
```
Comment (CIPop) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: I don't think we need to keep adding these comments. (It's part of our goals.)

File: src/System.Net.Requests/tests/AuthenticationManagerTest.cs [hunk 4]
```diff
@@ -12,29 +12,41 @@ namespace System.Net.Tests
 {
     public class AuthenticationManagerTest
     {
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "AuthenticationManager supported on NETFX")]
```
Comment (CIPop) [inline src/System.Net.Requests/tests/AuthenticationManagerTest.cs]: Is there a test that verifies this works on .Net Framework? Are we now porting all tests from .Net Framework here?

File: src/System.Net.Requests/tests/GlobalProxySelectionTest.cs [hunk 5]
```diff
@@ -38,6 +38,7 @@ public bool IsBypassed(Uri host)
             }
         }
 
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "dotnet/corefx #17842")]
```
Comment (CIPop) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: For consistency we should only use // TODO: #... <comment>

File: src/System.Net.Requests/src/System/Net/FtpWebRequest.cs (hunk 2 above)
Comment (davidsh) [inline src/System.Net.Requests/src/System/Net/FtpWebRequest.cs]: It is more of an implementation detail in `FtpWebRequest` with the control channel.

Comment from Desktop code:

>If KeepAlive is set to false, then the control connection to the server will be closed when the request completes

File: src/System.Net.Requests/tests/AuthenticationManagerTest.cs (hunk 4 above)
Comment (davidsh) [inline src/System.Net.Requests/tests/AuthenticationManagerTest.cs]: The goal right now is not to port all tests from .NET Framework.  The goal is to get the existing tests in CoreFx to work reliably.

File: src/System.Net.Requests/tests/GlobalProxySelectionTest.cs (hunk 5 above)
Comment (davidsh) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: >For consistency we should only use // TODO: #...

I've seen the issue number like this (in the SkipOnTargetFramework) in other libraries in CoreFx.

File: src/System.Net.Requests/src/System/Net/HttpWebRequest.cs (hunk 3 above)
Comment (davidsh) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: I can remove it but I thought it was helpful to future reviewers when they look at this code and try to figure out if NotSupport/NotImplemented was intentional or just a temporary thing. I will clean those up on next PR with other test fixes.

File: src/System.Net.Requests/src/System/Net/FtpWebRequest.cs (hunk 2 above)
Comment (CIPop) [inline src/System.Net.Requests/src/System/Net/FtpWebRequest.cs]: That doesn't actually answer my question.

The property is documented here: https://msdn.microsoft.com/en-us/library/system.net.ftpwebrequest.keepalive(v=vs.110).aspx
If the new default is "true" will CoreFX behave as documented?

File: src/System.Net.Requests/src/System/Net/HttpWebRequest.cs (hunk 3 above)
Comment (CIPop) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: I don't find the comment obvious in that case. Having a test that runs in all platforms and ensures the right exception would be preferable to a comment. (I thought I saw a test change already.)

Comment (davidsh) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: I'll remove the comment.

File: src/System.Net.Requests/src/System/Net/FtpWebRequest.cs (hunk 2 above)
Comment (davidsh) [inline src/System.Net.Requests/src/System/Net/FtpWebRequest.cs]: It is un-implemented in CoreFx.  

This is the comment in the code:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Requests/src/System/Net/FtpWebRequest.cs#L1585

> // We don't support connection pooling, so just silently ignore this.

**Comment by davidsh** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu14.04 Debug

**Pull request merged**